pytube/innertube.py

```diff
@@ -0,0 +1,120 @@
+"""This module is designed to interact with the innertube API.
+
+This module is NOT intended to be used directly by end users, as each of the
+interfaces returns raw results. These should instead be parsed to extract
+the useful information for the end user.
+"""
+# Native python imports
+import json
+from urllib import parse
+
+# Local imports
+from pytube import request
+
+
+class InnerTube:
+    """Object for interacting with the innertube API."""
+    @property
+    def base_url(self):
+        """Return the base url endpoint for the innertube API."""
+        return 'https://www.youtube.com/youtubei/v1'
+
+    @property
+    def base_data(self):
+        """Return the base json data to transmit to the innertube API."""
+        return {
+            'context': {
+                'client': {
+                    'clientName': 'WEB',
+                    'clientVersion': '2.20200720.00.02'
+                }
+            }
+        }
+
+    @property
+    def base_params(self):
+        """Return the base query parameters to transmit to the innertube API."""
+        return {
+            'key': 'AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8'
```

Comment:
Hey @tfdahlin sorry to comment on a merged PR, but out of interest where did this key come from? I have seen the `clientName` and `clientVersion` on the youtube html source but not the key...

Reply:
The key came from looking at xhr requests made by my browser

Reply:
It's easiest to trigger requests with this key if you e.g. scroll so that the browser needs to load additional content, such as more videos in a playlist/channel/search results

Reply:
Thanks!